lib/src/sql/array.rs

```diff
@@ -317,13 +318,18 @@ pub trait Uniq<T> {
 
 impl Uniq<Array> for Array {
 	fn uniq(mut self) -> Array {
-		for x in (0..self.len()).rev() {
-			for y in (x + 1..self.len()).rev() {
-				if self[x] == self[y] {
-					self.remove(y);
-				}
+		let mut set: HashSet<&Value> = HashSet::new();
+		let mut to_remove: Vec<usize> = Vec::new();
+		for i in 0..self.len() {
```

Comment:
Might be better style to use `for (i, item) in self.iter().enumerate() {...}` if possible, to avoid `&self[i]`